src/sentry/api/endpoints/internal_beacon.py

```diff
@@ -0,0 +1,59 @@
+import logging
+
+from django.conf import settings
+from rest_framework.response import Response
+
+from sentry import get_version, options
+from sentry.api.base import Endpoint
+from sentry.http import safe_urlopen
+from sentry.tasks.beacon import BEACON_URL
+
+logger = logging.getLogger("beacon")
+
+
+class InternalBeaconEndpoint(Endpoint):
+    permission_classes = ()
+
+    def post(self, request):
+        error = None
+        install_id = options.get("sentry:install-id")
+
+        if not settings.SENTRY_BEACON:
+            logger.info(
+                "beacon_metric.skipped", extra={"install_id": install_id, "reason": "disabled"}
+            )
+            return Response(status=204)
+
+        if settings.DEBUG:
+            logger.info(
+                "beacon_metric.skipped", extra={"install_id": install_id, "reason": "debug"}
+            )
+            return Response(status=204)
+
+        anonymous = options.get("beacon.anonymous") is not False
+
+        payload = {
+            "type": "metric",
+            "install_id": install_id,
+            "version": get_version(),
+            "anonymous": anonymous,
+        }
+
+        if not anonymous:
+            payload["admin_email"] = options.get("system.admin-email")
+
+        # Because this is used by the frontend, we want our frontend calls to be batched
+        # in order to reduce the number requests
+        data = request.data.get("data")
+        payload_data = [data] if data is not None else request.data.get("batch_data", [])
+
+        for payload_data_item in payload_data:
+            try:
+                safe_urlopen(BEACON_URL, json={**payload, "data": payload_data_item}, timeout=5)
+            except Exception:
+                logger.warning(
+                    "beacon_metric.failed", exc_info=True, extra={"install_id": install_id}
+                )
+                error = "Request failed"
+
+        return Response({"error": "Request failed"}, status=500) if error else Response(status=204)
```

Comment:
I know this would complicate things a bit but I think we should push these to the workers with a proper retry mechanism in case the self-hosted instance has external connectivity issues or simply our service is down.

Also, note to self: we may want to add a killswitch response to avoid hammering our mothership in case of emergencies. @JTCunning - I guess we can easily blackhole specific URLs for that? (so the `BEACON_URL` in this case?)

Reply:
I don't think it's worth the added complexity - the chances of external network issues or Sentry being down are pretty low, these metrics are also not as important as the normal beacon metrics either.

Reply:
My concern was more around conserving resources rather than data loss. A network issues usually manifests with a timeout which is quite costly as we'll be keeping a worker thread which is also linked to the front-end until we get a timeout, only to lose data. If this happens via 10s of users than we can easily starve the web instance. Self-hosted instances are usually frugal.

Ideally we'd just push the payload to a queue and respond immediately and then process the queue.

Reply:
Ah makes sense, sorry for the misunderstanding.

Reply:
No worries, my bad for not making the point clearer.